homeassistant/components/stt/__init__.py

```diff
@@ -193,9 +193,18 @@ async def async_process_audio_stream(
     @callback
     def check_metadata(self, metadata: SpeechMetadata) -> bool:
         """Check if given metadata supported by this provider."""
+        if metadata.language not in self.supported_languages:
+            language_matches = language_util.matches(
+                metadata.language,
+                self.supported_languages,
+            )
+            if language_matches:
+                metadata.language = language_matches[0]
```

Comment:
This side effect is a bit unexpected I think. Should this method really modify the input object?

Reply:
Fixed in https://github.com/home-assistant/core/pull/92002